org.eclipse.january.test/src/org/eclipse/january/dataset/MathsTest.java

```diff
@@ -2180,12 +2180,21 @@ public void testFloorRemainder() {
 		TestUtils.assertDatasetEquals(DatasetFactory.createFromObject(new float[] {-1.5f, -0.5f, -2, -1, 0, -1.5f, -0.5f, -2}),
 				Maths.floorRemainder(xa, -2.5f), true, ABSERRD, ABSERRD);
 	}
-
+	
+	@Test
+	public void testArctan2() {
+		Dataset a = DatasetFactory.createFromObject(new double[] { 4, 2, 6 });
+		Dataset b = DatasetFactory.createFromObject(new double[] { 1, 2, 3 });
+		TestUtils.assertDatasetEquals(
```

Comment:
Follow Peter's lead and make the same changes here, i.e use a loop to create your output instead of getDouble(0), getDouble(1), etc.